backend/src/test/java/gov/cdc/usds/simplereport/service/AddressValidationServiceTest.java

```diff
@@ -0,0 +1,80 @@
+package gov.cdc.usds.simplereport.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.smartystreets.api.exceptions.SmartyException;
+import com.smartystreets.api.us_street.Candidate;
+import com.smartystreets.api.us_street.Client;
+import com.smartystreets.api.us_street.Lookup;
+import com.smartystreets.api.us_street.Metadata;
+import gov.cdc.usds.simplereport.db.model.auxiliary.StreetAddress;
+import gov.cdc.usds.simplereport.service.errors.InvalidAddressException;
+import java.io.IOException;
+import java.util.ArrayList;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class AddressValidationServiceTest {
+
+  private AddressValidationService s;
+
+  @BeforeEach
+  public void setup() throws SmartyException, IOException {
+    Client client = mock(Client.class);
+    doNothing().when(client).send(isA(Lookup.class));
+    s = new AddressValidationService(client);
+  }
+
+  @Test
+  void throwsExceptionWhenNoAddressesAreFound() {
+    Lookup lookup = mock(Lookup.class);
+    when(lookup.getResult()).thenReturn(new ArrayList<Candidate>());
+
+    InvalidAddressException caught =
+        assertThrows(
+            InvalidAddressException.class,
+            () -> {
+              s.getValidatedAddress(lookup);
+            });
+    assertEquals("Address entered could not be validated", caught.getMessage());
+  }
+
+  @Test
+  void setsCountyOnStreetAddress() throws SmartyException, IOException {
+    ArrayList<Candidate> results = new ArrayList<Candidate>();
+    results.add(getMockResult());
+    Lookup lookup = mock(Lookup.class);
+    when(lookup.getResult()).thenReturn(results);
```

Comment:
```suggestion
    Lookup lookup = mock(Lookup.class);
    when(lookup.getResult()).thenReturn(List.of(getMockResult()));
```

Reply:
This requires casting the List to an ArrayList and then seems to be causing the test to fail with 
```
E.g. thenReturn() may be missing.
Examples of correct stubbing:
    when(mock.isOk()).thenReturn(true);
    when(mock.isOk()).thenThrow(exception);
    doThrow(exception).when(mock).someVoidMethod();
Hints:
 1. missing thenReturn()
 2. you are trying to stub a final method, which is not supported
 3. you are stubbing the behaviour of another mock inside before 'thenReturn' instruction is completed
 ```

Reply:
Well, that's a part foul on whoever wrote that interface, but not much for us to do about it. Sorry for the red herring!